### PR TITLE
Chore: Create `handleReduceMotion` util function to manage animations

### PR DESCRIPTION
**Pull request by eledobleefe**
**What is this feature?**
We create a util function to handle animations and how it behaves when `prefers-reduce-motion` is enabled.

**Why do we need this feature?**
To allow devs to manage animations when the user has `prefers-reduce-motion` enabled more easily.

**Who is this feature for?**
Devs creating animations

**Which issue(s) does this PR fix?**:

Fixes #
Related to #85450

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by ashharrison90 — COMMENTED** [review]
this needs exposing in `utils/index` so that it's exposed properly in `@grafana/ui`

**Comment by eledobleefe** [comment]
> this needs exposing in `utils/index` so that it's exposed properly in `@grafana/ui`

Missed that. Thanks for shouting it up.

File: packages/grafana-ui/src/utils/handleReduceMotion.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { CSSInterpolation } from '@emotion/css';
+
```
Comment (ashharrison90) [inline packages/grafana-ui/src/utils/handleReduceMotion.ts:2]: think we could also add some JSDoc here about how to use this function and what the args are

File: packages/grafana-ui/src/utils/handleReduceMotion.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+export const handleReduceMotion = (styles: CSSInterpolation, reduceMotionStyles?: CSSInterpolation) => {
```
Comment (ashharrison90) [inline packages/grafana-ui/src/utils/handleReduceMotion.ts:3]: minor nit, think we should use `reducedMotion` always to match the media query name
```suggestion
export const handleReducedMotion = (styles: CSSInterpolation, reducedMotionStyles?: CSSInterpolation) => {
```

File: packages/grafana-ui/src/utils/handleReduceMotion.ts
Comment (ashharrison90) [inline packages/grafana-ui/src/utils/handleReduceMotion.ts:1]: same with the file name, `handleReducedMotion`

**Review by ashharrison90 — APPROVED** [review]
couple of nits, otherwise lgtm 👍

File: packages/grafana-ui/src/utils/handleReduceMotion.ts (hunk 2 above)
Comment (eledobleefe) [reply packages/grafana-ui/src/utils/handleReduceMotion.ts:3]: I think I always write it wrong 😅

File: packages/grafana-ui/src/utils/handleReducedMotion.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { CSSInterpolation } from '@emotion/css';
+
+/**
+ * @param styles - Regular styles.
```
Comment (ashharrison90) [inline packages/grafana-ui/src/utils/handleReducedMotion.ts:4]:
```suggestion
 * @param styles - Styles to apply when no `prefers-reduced-motion` preference is set.
```

File: packages/grafana-ui/src/utils/handleReducedMotion.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ * @param reducedMotionStyles - Styles to apply when `prefers-reduced-motion` is enabled.
+ * If @param reducedMotionStyles is not provided, there won't be any animation or transition shown.
```
Comment (ashharrison90) [inline packages/grafana-ui/src/utils/handleReducedMotion.ts:6]:
```suggestion
 * Applies one of `styles` or `reducedMotionStyles` depending on a users `prefers-reduced-motion` setting.
```

**Review by ashharrison90 — APPROVED** [review]
think this is slightly clearer maybe? 🤔

File: packages/grafana-ui/src/utils/handleReducedMotion.ts (hunk 4 above)
Comment (eledobleefe) [reply packages/grafana-ui/src/utils/handleReducedMotion.ts:6]: But we are not explaining what happens if `reducedMotionStyles` is not set, right? If there is no value then the animation, when `prefers-reduced-motion` is enabled, won't be shown.

Comment (eledobleefe) [reply packages/grafana-ui/src/utils/handleReducedMotion.ts:6]: yeah! Like it 😉

**Review by ashharrison90 — APPROVED** [review]
looks good 🥳

**Pull request merged**